### DISCUSSION TITLE
Problems wot connect on live port with some encoder software and with telnet

### DISCUSSION THREAD

**Discussion by caf2022** [savonet/liquidsoap#2399] [User Support]
Hi, I'm having problem on some servers(not all) when try to connect to live broadcast with Opticodec nothing occurs, the connection doesn't even reach the server, but if I use radioboss the connection is successfully.

Another problem with the same server that have the issue above is the telnet, I can connect to the telnet but I can't run commands,  keep thinking until timeout appears:

```
# telnet localhost 1001
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
help [it takes a while]
Connection closed by foreign host.
```

All the servers have the same configurarion, no firewall, no hosting control panel, it's Centos 7 clean but on some server I'm getting this issue.

Any help is welcome, thanks

**Comment by toots** [comment]
Hi,

I would love to help but I would need more information:
* What version of liquidsoap are you using?
* What do you mean by connecting? What are you connecting to?

Can you provide script extract? Logs extract if they are relevant.

**Comment by caf2022** [comment]
Hi, thanks for your response.

**What version of liquidsoap are you using?**

Tried 2.0.4-1 and 2.0.5 with opam

**What do you mean by connecting? What are you connecting to?**

The live broadcast not connect, I'm trying to connect to live broadcast using encoders software, with radioboss connect and send the stream to the liquidsoap, but the opticodec and some others encoders can't connect, keep trying to connect and on the liquidsoap not register any log, is like if the liquidsoap was offline and the encoder software cand research the port, but like I said, others stream software can connect fine.

On the logs not show any information about errors, only startup and playlist files.

The strangest thing is that it works on another server, and the 2 servers are the same, same OS, same configuration, no firewall, no rules in iptables, nothing special installed, only the pure OS.

```
tcp        0      0 127.0.0.1:3350          0.0.0.0:*               LISTEN      19558/liquidsoap [TELNET]
tcp        0      0 0.0.0.0:39936           0.0.0.0:*               LISTEN      19558/liquidsoap [LIVE INPUT HARBOR]
tcp        0      0 0.0.0.0:39937           0.0.0.0:*               LISTEN      19558/liquidsoap
tcp        0    178 172.82.129.42:45312     172.82.129.42:11599     ESTABLISHED 19558/liquidsoap [SHOUTCAST]

```
on the same servers where this problem occurs, there is also a problem with the telnet that connects but does not execute commands, it get timeout when I try to execute a command

**Comment by caf2022** [comment]
THe encoder keep trying to connect, but if I use other software, the connection is successfully, but, the opticodec run fine on other server with liquidsoap and the same configuration as I said.

thank you very much for the help, i appreciate it very much

**Comment by toots** [comment]
Shoutcast protocol needs two ports to be accessible:
* The configured port `N`
* The next port `N+1`
Did you make sure that both ports are open? Here that should be `39936` and `39937`.

**Comment by caf2022** [comment]
Yes, look:

```
tcp        0      0 0.0.0.0:39936           0.0.0.0:*               LISTEN      19558/liquidsoap
tcp        0      0 0.0.0.0:39937           0.0.0.0:*               LISTEN      19558/liquidsoap
```

`dj = amplify(override="replay_gain", streaming_gain_dj,input.harbor(id="aovivo", icy=true,"/",port=39936,auth=dj_auth, icy_metadata_charset = "UTF-8", metadata_charset = "UTF-8", on_connect=fun (_) -> dj_conectado("on",porta), on_disconnect=fun () -> dj_desconectado("off",porta)))`

**Reply by toots** [reply]
Do you have a firewall installed?

**Comment by caf2022** [comment]
No firewall, no iptables rules, no DDoS filter.

And like I said, if I use other software like Radioboss for example, works fine, but the opticodec can't work, BUT, only on some servers(like 3), on the others server is running fine with any software and all servers have the same configuration, same OS, same versions, all identical.

```
# iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

```

**Comment by caf2022** [comment]
this is very strange, I checked with data center support too and they didn't find any errors or blockages on the network

**Comment by caf2022** [comment]
I changed the log to 4 and now showing this when try to connect:

`2022/05/20 16:07:03 [harbor:4] New client on port 39190: 186.80.28.41`

But keep on this, the broadcast software don't show that have connected, keep trying to connect and on the LS logs not show anymore.

More logs:

```
2022/05/20 16:06:46 [harbor:4] Opening port 39190 with icy = false
2022/05/20 16:06:46 [harbor:4] Opening port 39191 with icy = true
2022/05/20 16:06:46 [harbor:3] Adding mountpoint '/' on port 39190
```

**Comment by caf2022** [comment]
Solved with OS reload and installing Ubuntu 20.x something on the Centos 7 template causing this.

**Reply by toots** [reply]
Ok thanks for letting us know!
